src/modules/game/game.service.ts

```
@@ -153,56 +227,28 @@ export class GameService {
-      throwHttpError(ErrorCode.GAME_INVALID);
-    }
-
-    // Check if there are moves to undo
-    if (game.moves.length === 0) {
-      throwHttpError(ErrorCode.INVALID_MOVE); // or custom "NO_MOVES_TO_UNDO"
-    }
-
-    // Optional: enforce turn ownership (only the player who just moved can undo)
     const chess = new Chess(game.currentFen);
-    const lastMove = game.moves[game.moves.length - 1];
+    const lastMove = game.moves.pop();
```

ESLint

[error] 'lastMove' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)